src/api/store/vendor.py

```diff
@@ -196,13 +213,28 @@ def copy_vendor(self, vendor_id) -> Tuple[Vendor, MassEnergizeAPIError]:
       vendor: Vendor = Vendor.objects.get(id=vendor_id)
       if not vendor:
         return CustomMassenergizeError(f"No vendor with id {vendor_id}")
-        
-      vendor.pk = None
-      vendor.is_published = False
-      vendor.is_verified = False
-      vendor.name = f"{vendor.name}-Copy-{random.randint(1,100000)}"
-      vendor.save()
-      return vendor, None
+
+      old_tags = vendor.tags.all()
+      new_vendor = vendor
+
+      new_vendor.pk = None
+      new_vendor.is_published = False
+      new_vendor.is_verified = False
+
+      # the copy will have "-Copy" appended to the name; if that already exists, delete it first
+      new_name = vendor.name + "-Copy"
+      check = Vendor.objects.filter(name=new_name).first()
+      if check:
+        check.delete()
```

Comment:
Hmmmm just checked it out.  This approach also seems a bit dangerous.  The vendor which is the previous copy in this case could be linked to so many other models and instances.  Ideally, instead of deleting an old copy and basically creating it as a new one, it may be better to rather
* Copy over the latest updates into the copy that already exists and return the old copy -- this way we are not losing any data and are not making more copies with the same name.
* Make the copies have numbers but not randomized.   Maybe they can be in a progression.  


I really recommend leaving the random number thing in and then we can create a separate PR that tackles the core of the problem which is "How do we support creating copies without having orphan copies that bloat our system?"  
That seems like a very good chunk of problem to solve in isolation rather than squeezing it into this one.

Reply:
I've implemented the first method.